staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers_test.go

```diff
@@ -178,25 +188,29 @@ func TestParseRuntimeConfig(t *testing.T) {
 			expectedAPIConfig: func() *serverstore.ResourceConfig {
 				config := newFakeAPIResourceConfigSource()
 				config.DisableVersions(extensionsapiv1beta1.SchemeGroupVersion)
+				// disabling groups of APIs removes the individual resource preferences from the default
+				config.RemoveMatchingResourcePreferences(matchAllExplicitResourcesForFake)
 				return config
 			},
 			err: false,
 		},
 		{
-			// disable a non-extensions resource
+			name: "disable-a-no-extensions-resources",
 			runtimeConfig: map[string]string{
 				"apps/v1/deployments": "false",
 			},
 			defaultResourceConfig: func() *serverstore.ResourceConfig {
 				return newFakeAPIResourceConfigSource()
 			},
 			expectedAPIConfig: func() *serverstore.ResourceConfig {
-				return newFakeAPIResourceConfigSource()
+				config := newFakeAPIResourceConfigSource()
+				config.DisableResources(appsv1.SchemeGroupVersion.WithResource("deployments"))
```

Comment:
This is now allowed since we're going to need it to enable particular beta APIs.